lib/manager/html/extract.ts

```diff
@@ -0,0 +1,46 @@
+import { PackageFile, PackageDependency } from '../common';
+import { DATASOURCE_CDNJS } from '../../constants/data-binary-source';
+import { cloudflareUrlRegex } from '../cdnurl/extract';
+
+const regex = /<\s*(script|link)\s+[^>]*?\/?>/i;
+
+export function extractDep(tag: string): PackageDependency | null {
+  const match = cloudflareUrlRegex.exec(tag);
+  if (match) {
+    const { depName, currentValue, asset } = match.groups;
+    return {
+      datasource: DATASOURCE_CDNJS,
+      depName,
+      lookupName: `${depName}/${asset}`,
+      currentValue,
+    };
+  }
+  return null;
+}
+
+export function extractPackageFile(content: string): PackageFile {
+  const deps: PackageDependency[] = [];
+
+  let rest = content;
+  let match = regex.exec(rest);
```

Comment:
We know browsers are liberal on html errors, but how well the parser and our code would handle them? Could it also handle templates like php or something?

Reply:
Are you confident that the current regex will catch all `<script>` instances and its contents? I guess we add a few test cases and keep extending that in future if we find some that don't work?

Reply:
Okay, also I'll search for error-tolerant html parser and will use it if possible

Reply:
There are edge-cases we definitely can't foresee

Reply:
I am open to your idea though. There definitely could be cases where non-html files such as templates or partials contain scripts with integrity hashes. So by just looking for script instead of parsing the whole html, we maybe satisfy more use cases.